src/raven.js

```diff
@@ -1611,7 +1604,16 @@ Raven.prototype = {
             url = url.replace(/^https?:/, '');
 
             // onreadystatechange not supported by XDomainRequest
-            request.onload = handler;
+            if (opts.onSuccess) {
+                request.onload = opts.onSuccess;
+            }
+            if (opts.onError) {
+                request.onerror = function () {
+                    var err = new Error('Sentry error code: XDomainRequest');
```

Comment:
I quickly experimented w/ having a helper `onerror` function that could be called from either the `XDomainRequest` handler or `onreadystatechange`, but it quickly ended up taking up more code so nm ...